app/src/main/java/com/example/sridharjajoo/newsapp/core/Headline/HeadlineFragment.java

```diff
@@ -86,6 +90,15 @@ private void loadNewsArticles() {
                 setRecyclerView(articlesList);
             }
         });
+
+        pullToRefresh.setOnRefreshListener(() -> {
+            if (!Utils.hasNetwork()) {
+                Toast.makeText(getActivity(), "Network not available!", Toast.LENGTH_SHORT).show();
+            }
```

Comment:
Add "return;" here otherwise even though the network isn't there then the statements outside the if will be executed